homeassistant/components/spc/__init__.py

```diff
@@ -16,61 +14,60 @@
 CONF_API_URL = "api_url"
 
 DOMAIN = "spc"
-DATA_API = "spc_api"
+DATA_KEY = "spc"
 
 SIGNAL_UPDATE_ALARM = "spc_update_alarm_{}"
 SIGNAL_UPDATE_SENSOR = "spc_update_sensor_{}"
-
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_WS_URL): cv.string,
-                vol.Required(CONF_API_URL): cv.string,
-            }
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
+PLATFORMS = ["alarm_control_panel", "binary_sensor"]
 
 
 async def async_setup(hass, config):
+    """Old way to set up SPC devices."""
+    return True
+
+
+async def async_setup_entry(hass, entry):
     """Set up the SPC component."""
 
-    async def async_upate_callback(spc_object):
+    await async_setup_internal(hass, entry)
+
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, platform)
+        )
+
+    return True
+
+
+async def async_setup_internal(hass, entry):
+    """Connect the API client."""
+    config = entry.data
 
+    async def async_upate_callback(spc_object):
         if isinstance(spc_object, Area):
             async_dispatcher_send(hass, SIGNAL_UPDATE_ALARM.format(spc_object.id))
         elif isinstance(spc_object, Zone):
             async_dispatcher_send(hass, SIGNAL_UPDATE_SENSOR.format(spc_object.id))
 
+    hass.data.setdefault(DATA_KEY, {})
```

Comment:
This sets a top-level key in `hass.data`, which is a no-no – put it under your `DOMAIN`.

Reply:
Actually DATA_KEY is the domain name.

Reply:
`DOMAIN` and `DATA_KEY` have the same value; why not just use `DOMAIN`?

https://github.com/home-assistant/core/blob/e09faed161ab7e9f1844e6ae2ab3c3cd00aa4a33/homeassistant/components/spc/__init__.py#L16

Reply:
It was separated like that in a couple of other integrations that I used "for inspiration". But I agree. Removed.